partition pruning fct_ga4__pages

dgitis opened the pull request:
## Description & motivation

The fct_ga4__pages model was missing is_incremental checks causing it to process more data than needed. This PR fixes that.

## Checklist
- [ y] I have verified that these changes work locally
- [n/a ] I have updated the README.md (if applicable)
- [n/a ] I have added tests & descriptions to my models (and macros if applicable)
- [ y] I have run `dbt test` and `python -m pytest .` to validate existing tests

adamribaudo-velir reviewed (changes requested):
I'm getting the following error:

```
10:51:53  Completed with 1 error and 0 warnings:
10:51:53
10:51:53  Database Error in model fct_ga4__pages (models\marts\core\fct_ga4__pages.sql)
10:51:53    Syntax error: Unexpected ")" at [41:33]
10:51:53    compiled Code at target\run\ga4\models\marts\core\fct_ga4__pages.sql
```

Also, you need to assign `partitions_to_replace`

See the config block in `fct_ga4__sessions_daily` as an example model that supports both dynamic and static incremental runs.

dgitis commented:
I fixed the config block. 

I can't reproduce the Database error. I fixed exactly that error in 5a445d21ada96b61834ea390b2a71f249c189e99. Maybe you somehow got the older version of this branch when you reviewed.
